assignment/poker/PokerEvaluator.md

```diff
@@ -0,0 +1,167 @@
+comparison, for example, `QQQ88 > 999KK`, `KKK77 > QQQJJ` and `KKK77 > KKK66`.
+
+When comparing `Straight`-s, the `A2345` `Straight` is the weakest one and the `TJQKA` one the strongest one,
+for example, `23456 > A2345` and `TJQKA > 9TJQK`.
+
+If any of these rules are not clear, please consult [Poker hand and categories](https://en.wikipedia.org/wiki/List_of_poker_hands#Hand-ranking_categories) and
+the game rules of the specific poker games.
+
+## Implementation
+
+### Input and Output
+
+Your code should read from standard input (`stdin`) and write to standard output (`stdout`) until `EOF` is 
+reached (see [standard streams](https://en.wikipedia.org/wiki/Standard_streams)).
+
+The program will be tested by piping input to `stdin` and piping output to `stdout`, with up to 100,000 
+lines of input per a single run.
+
+There will be 1 test case per 1 line of input, your program should produce a corresponding 1 line of output
+for each line of input, even in case of errors. In case the input line is invalid or not supported by 
+your solution, output a clear and easy to understand error message, prefixed with `Error: `.
+
+Please do NOT output extra text or expect extra input besides the lines with test cases.
+
+#### Input of Test Cases
+
+The input is to be read from the standard input, with one test case per line:
+
+```
+<game-type> [<5 board cards>] <hand 1> <hand 2> <...> <hand N>
+```
+
+...where: 
+
+* `game-type` specifies the game type for this test case, one of:
+  * `texas-holdem` - for Texas Hold'em
+  * `omaha-holdem` - for Omaha Hold'em
+  * `five-card-draw` - for Five Card Draw
+
+* `<5 board cards>` is a 10 character string where each 2 characters encode a card, only used for Texas and 
+Omaha Hold' ems
+ 
+* `<hand X>` is a 4, 8 or 10 character string (depending on game type) where each 2 characters encode a card
+* `<card>` is a 2 character string with the first character representing the rank 
+(one of `A`, `K`, `Q`, `J`, `T`, `9`, `8`, `7`, `6`, `5`, `4`, `3`, `2`) and the second character representing 
+the suit (one of `h`, `d`, `c`, `s`). Jokers are not used. 
+
+#### Output of Solutions
+
+The output is to be written to standard output using the format:
+
+```
+<hand block 1> <hand block 2> <...> <hand block n>
+```
+... where:
+
+* `<hand block 1>` is the hand block with the weakest value
+* `<hand block 2>` is the hand block with the second weakest value
+* ... and so forth.
+* `<hand block n>` is the hand block with the strongest value
+
+Each hand block consists of one or multiple hands (each represented by 4, 8 or 10 character string, depending 
+on game type, with 2 characters to encode a card) with equal hand value.
+
+In case there are multiple hands with the same value on the same board they should be ordered alphabetically 
+and separated by `=` signs.
+
+The order of the cards in each hand should remain the same as in the input, e.g., don't reorder `2h3s` into 
+`3s2h`.
+
+Example input:
```

Comment:
Move this higher to the Input section?

Reply:
I see it's coupled with the output. Still mb it's worth to copy-paste it above?

Reply:
Introduced a header, is it better?